packages/hoppscotch-backend/src/user-collection/user-collection.service.ts

```diff
@@ -1182,9 +1195,55 @@ export class UserCollectionService {
       userID,
       collection.right.parentID,
       reqType,
+      false,
     );
     if (E.isLeft(result)) return E.left(result.left as string);
 
     return E.right(true);
   }
+
+  /**
+   * Generates a JSON containing all the contents of a collection
+   *
+   * @param collection Collection whose details we want to fetch
+   * @returns A JSON string containing all the contents of a collection
+   */
+  private async fetchCollectionData(collectionID: string) {
```

Comment:
Noticed the return type wasn't getting inferred correctly and might want to specify it.

```suggestion
  private async fetchCollectionData(
    collectionID: string,
  ): Promise<E.Left<string> | E.Right<UserCollectionDuplicatedData>> {
```